### PR TITLE
Clean up some code relating to paths and saving status, preparing for Save to Cloud features

### PR DESCRIPTION
**Pull request by cbjeukendrup**
Closes: #10552
Resolves: #10288
Resolves: #10631

The solution with "SaveLocation" was unnecessarily complicated, so we keep it simple and identify scores by an `io::path`. There will be a hidden directory where cloud projects are stored/cached. We distinguish cloud scores by checking if their path is inside that hidden directory.

### CODE REVIEW

File: src/engraving/libmscore/masterscore.h [hunk 1]
```diff
@@ -222,6 +226,15 @@ class MasterScore : public Score
     IFileInfoProviderPtr fileInfo() const;
     void setFileInfoProvider(IFileInfoProviderPtr fileInfoProvider);
 
+    bool isNewlyCreated() const { return m_isNewlyCreated; }
```
Comment (RomanPudashkin) [inline src/engraving/libmscore/masterscore.h:229]: Please, move the implementation to cpp

File: src/project/internal/projectautosaver.cpp [hunk 2]
```diff
@@ -84,19 +82,40 @@ void ProjectAutoSaver::removeProjectUnsavedChanges(const io::path& projectPath)
...
+        return projectAutoSavePath;
+    }
+
     return io::filename(projectAutoSavePath, false);
 }
 
 mu::io::path ProjectAutoSaver::projectAutoSavePath(const io::path& projectPath) const
 {
-    return projectPath + AUTOSAVE_SUFFIX;
+    return projectPath + "." + AUTOSAVE_SUFFIX;
```
Comment (RomanPudashkin) [inline src/project/internal/projectautosaver.cpp:94]: You can reuse appendingSuffix here

File: src/framework/global/io/path.cpp [hunk 3]
```diff
@@ -49,6 +54,24 @@ bool path::empty() const
     return m_path.isEmpty();
 }
 
+path path::appendingComponentsOf(const path& other) const
```
Comment (RomanPudashkin) [inline src/framework/global/io/path.cpp:57]: Maybe just appendingComponent? Its name should be similar with appendingSuffix

File: src/framework/global/io/path.cpp [hunk 4]
```diff
@@ -49,6 +54,24 @@ bool path::empty() const
...
+{
+    if (m_path.endsWith('/') || other.m_path.startsWith('/')) {
+        return m_path + other.m_path;
+    }
+
+    return m_path + '/' + other.m_path;
+}
+
+path path::appendingSuffix(const QString& suffix) const
```
Comment (RomanPudashkin) [inline src/framework/global/io/path.cpp:66]: You can use io::path for suffixes

**Review by RomanPudashkin — APPROVED** [review]

File: src/project/internal/projectautosaver.cpp (hunk 2 above)
Comment (cbjeukendrup) [reply src/project/internal/projectautosaver.cpp]: Nice one, thanks! Fixed (other comments too).

**Comment by DmitryArefiev** [comment]
Tested #10288 #10631 on Win10, Mac12, LinuxMint 20.2 - FIXED      

Please merge into master

@cbjeukendrup #10589 still occurs..

**Comment by cbjeukendrup** [comment]
> #10589 still occurs..

Ah, yes, that's right. It will be fixed in the next PR in the series: #10662.

**Pull request merged**